src/controllers/user.js

```diff
@@ -57,11 +57,30 @@ export const getAll = async (req, res) => {
 }
 
 export const updateById = async (req, res) => {
-  const { cohort_id: cohortId } = req.body
-
-  if (!cohortId) {
-    return sendDataResponse(res, 400, { cohort_id: 'Cohort ID is required' })
+  const id = parseInt(req.params.id)
+  const userToUpdate = await User.fromJson(req.body)
+  userToUpdate.id = id
+
+  // Update cohortId and role to userToUpdate (only if logged in role is Teacher)
+  // otherwise, keep the existing values of the user
+  if (req.user.role === 'TEACHER') {
```

Comment:
this check is for middleware auth.
all validation regaring user if possible should be handled in middleware and functions to be called from route

Reply:
I can move it there, in this case I would then have to modify the request body in the middleware, in order for it to include the given variables for role and cohortId when the request user is a TEACHER, or to find the already existing fields for role and cohortId when the request user is a STUDENT. Which do you/Magnus/Nigel prefer?

Short explanation about what this check does:
- Only a teacher can modify the role and cohortId fields.
- What the backend currently does with this implementation, is that it will accept all fields regardless of if it STUDENT or TEACHER, and only modify the role and cohortId fields if the role is TEACHER